src/main/java/seedu/address/model/person/Bloodtype.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's bloodtype in ORGANice.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBloodtype(String)}
+ */
+public class Bloodtype {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Bloodtype should only have A, B, O or AB."
+                    + "Positive bloodtypes will have a '+' behind, it should not be blank";
+    public static final String VALIDATION_REGEX = "\\d+";
+    public final String value;
+
+    /**
+     * Constructs a {@code Bloodtype}.
+     *
+     * @param bloodtype A valid bloodtype.
+     */
+    public Bloodtype(String bloodtype) {
+        requireNonNull(bloodtype);
+        checkArgument(isValidBloodtype(bloodtype), MESSAGE_CONSTRAINTS);
+        value = bloodtype;
+    }
+
+    /**
+     * Returns true if a given string is a valid bloodtype.
+     */
+    public static boolean isValidBloodtype(String test) {
```

Comment:
Can your isValidBloodType return false for inputs that are not any specific blood type? Refer to William's PR on the Type.java class for an example.

Reply:
Done, refer to lastest commit